homeassistant/components/seventeentrack/config_flow.py

```diff
@@ -0,0 +1,105 @@
+"""Adds config flow for 17track.net."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from py17track import Client as SeventeenTrackClient
+from py17track.errors import SeventeenTrackError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.schema_config_entry_flow import (
+    SchemaFlowFormStep,
+    SchemaOptionsFlowHandler,
+)
+
+from .const import (
+    CONF_SHOW_ARCHIVED,
+    CONF_SHOW_DELIVERED,
+    DEFAULT_SHOW_ARCHIVED,
+    DEFAULT_SHOW_DELIVERED,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+OPTIONS_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_SHOW_ARCHIVED, default=DEFAULT_SHOW_ARCHIVED): bool,
+        vol.Optional(CONF_SHOW_DELIVERED, default=DEFAULT_SHOW_DELIVERED): bool,
+    }
+)
+OPTIONS_FLOW = {
+    "init": SchemaFlowFormStep(OPTIONS_SCHEMA),
+}
+
+USER_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class SeventeenTrackConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """17track config flow."""
+
+    VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> SchemaOptionsFlowHandler:
+        """Get options flow for this handler."""
+        return SchemaOptionsFlowHandler(config_entry, OPTIONS_FLOW)
+
+    async def _async_validate_input(self, user_input):
+        """Validate the user input allows us to connect."""
+
+        session = aiohttp_client.async_get_clientsession(self.hass)
+
+        client = SeventeenTrackClient(session=session)
+
+        try:
+            login_result = await client.profile.login(
+                user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+            )
+
+            if not login_result:
+                _LOGGER.error("Invalid username and password provided")
+                return {"base": "invalid_credentials"}
+        except SeventeenTrackError as err:
+            _LOGGER.error("There was an error while logging in: %s", err)
+            return {"base": "cannot_connect"}
+
+        return {}
+
+    async def async_step_user(
```

Comment:
This step is missing a check for uniqueness. Can we fetch some kind of value from 17track that could act as unique identifier? Otherwise we should use `async_abort_entries_match` to compare the new user input with existing entities.

Can users change their username in the 17track app?